### PR TITLE
Refine complex_graph `regex_matches` partial suppressions

### PR DESCRIPTION
**Pull request by EliahKagan**
This builds on #1635 along the lines of the suggestion in https://github.com/GitoxideLabs/gitoxide/issues/1622#issuecomment-2530531958, by:

- Updating the comment about [#1622](https://github.com/GitoxideLabs/gitoxide/issues/1622), including by adding links to Git mailing list posts by Aarni Koskela, who discovered the bug that turns out to be the cause of this, and Patrick Steinhardt, who analyzed the bug and wrote a fix (currently in testing); and [#1622 (comment)](https://github.com/GitoxideLabs/gitoxide/issues/1622#issuecomment-2529580735), which summarizes that and reports on its connection to [#1622](https://github.com/GitoxideLabs/gitoxide/issues/1622).

- Narrowing the partial suppression of the failing test code (which consists of conditionally using `parse_spec_no_baseline` instead of `parse_spec` in some assertions) so that it is only done if Git is at one of the versions that is known to be affected.

  If any future Git versions are affected, such as by the currently cooking patch not being merged as soon as I expect, then this could potentially fail on CI again. But that is something we would probably want to find out about.

### CODE REVIEW

File: gix/tests/gix/revision/spec/from_bytes/regex.rs [hunk 1]
```diff
@@ -95,11 +95,19 @@ mod find_youngest_matching_commit {
...
+        // In Git 2.47.0 (and 2.47.1), the traversal order differs, so some `parse_spec` assertions
+        // fail. This is a Git bug with a forthcoming fix. For now, we use `parse_spec_no_baseline`
+        // for them when tests are run that way with known-affected Git versions. For details, see:
+        //
+        //  - https://lore.kernel.org/git/Z1LJSADiStlFicTL@pks.im/T/
+        //  - https://lore.kernel.org/git/Z1LtS-8f8WZyobz3@pks.im/T/
+        //  - https://github.com/GitoxideLabs/gitoxide/issues/1622#issuecomment-2529580735
+        let skip_some_baselines = is_ci::cached()
+            && std::env::var_os("GIX_TEST_IGNORE_ARCHIVES").is_some()
+            && ((2, 47, 0)..(2, 47, 2)).contains(&gix_testtools::GIT_VERSION);
```
[the remark this answers is not in the document]
Comment (Byron) [reply gix/tests/gix/revision/spec/from_bytes/regex.rs:108]: LGTM!

Comment (Byron) [inline gix/tests/gix/revision/spec/from_bytes/regex.rs:108]: I really love this line - a `.contains()` check on a range between tuples!

**Review by Byron — APPROVED** [review]
This is the perfect implementation to solve the problem at hand!

I had something much more simplistic in mind, but also something that would have been very broken :D.

**Comment by EliahKagan** [comment]
[The pure-rust-build job is stalled](https://github.com/GitoxideLabs/gitoxide/actions/runs/12258694383/job/34199066273#step:4:8) on the `apt-get update` command in the container. This didn't happen in my fork, and rerunning that job in my fork a couple times [doesn't produce it](https://github.com/EliahKagan/gitoxide/actions/runs/12258693573/job/34202360490). So I recommend canceling and rerunning that job here. I expect that it will pass if that is done.

This occasionally happens locally but I don't think I've seen it on CI before. Maybe using the `us.` country mirror would make it less likely (since I think the GitHub-hosted runners are usually deployed in regions in the US). This should be to do. but hard to test effectively if the problem occurs only rarely. It may be possible to observe whether that makes things faster, though. Anyway, whether or not such a change is made, I don't recommend it be done as part of this PR.

**Pull request merged**